qiskit/quantum_info/operators/unitary.py

```diff
@@ -0,0 +1,250 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+# pylint: disable=invalid-name,assignment-from-no-return
+
+"""
+Tools for working with Unitary Operators.
+
+A simple unitary class and some tools.
+"""
+import copy
+import numpy
+import sympy
+from qiskit.exceptions import QiskitError
+from qiskit.circuit.gate import Gate
+from qiskit.dagcircuit import DAGCircuit
+
+
+class Unitary(Gate):
+    """Class for representing unitary operators"""
+
+    def __init__(self, matrix, *qargs, label=None, validate=True, rtol=1e-5, atol=1e-8):
+        """
+        Create unitary.
+
+        Args:
+            matrix (numpy.ndarray or list(list)): unitary matrix representation.
+            qargs (list(tuple(QuantumCircuit, int))): qubits to apply operation to.
+            label (str): identifier hint for backend
+            validate (bool): whether to validate unitarity of matrix when supplied
+                either here or with Unitary.matrix.
+            rtol (float): relative tolerance (see numpy.allclose).
+            atol (float): absolute tolerance (see numpy.allclose).
+        """
+        self._matrix = None
+        self._label = None
+        self._validate = validate
+        self._rtol = rtol
+        self._atol = atol
+        self._n_qubits = None
+        self._dimension = None
```

Comment:
shouldn't `_n_qubits` and `_dimension` be populated here based on the passed in matrix?

Reply:
This is actually done with the passed in matrix in the `matrix` method. These are only here because I thought the linter was warning me about setting attributes in the `matrix` method which hadn't been defined in the `__init__`.